### PR TITLE
interpolate backsector->heightsec

### LINKED ISSUE #1137: Sector movment doesn't render somtimes using Boom fake floors (Line action 242)
This happens on the latest release (Woof! 11.3.0), It doesn't happen on the prior release (11.2.0). It also doesn't happen with Dsda-doom 0.26.2 or Eternity 4.02. 
When the sector informing the fake height of a sector moves the projected height is supposed to move with it. However, Woof will sometimes not render this change. This only happens if some section of the map is being completely occluded by the fake floor. It will render correctly if the player sees in or beyond the fake floor sector. This is kind of hard to communicate verbally, so here's a demo wad. The blood wall should appear to lower when you walk forward.
[woofrender.zip](https://github.com/fabiangreffrath/woof/files/11995943/woofrender.zip)

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1137

### CODE REVIEW

**Review by fabiangreffrath — APPROVED** [review]
Wow, that was fast, thank you! Is this compatible with the fixes to all the other issues that have been posted before?

File: src/r_bsp.c [hunk 1]
```diff
@@ -383,6 +383,11 @@ static void R_AddLine (seg_t *line)
   //      should already be interpolated.
   R_MaybeInterpolateSector(backsector);
 
+  if (backsector->heightsec != -1)
```
Comment (fabiangreffrath) [inline src/r_bsp.c:386]: Hm, could you maybe integrate this into `R_FakeFlat()`? It contains the same check for `backsector->heightsec != -1`. Just an idea. 😉

Comment (rfomin) [reply src/r_bsp.c:386]: > Is this compatible with the fixes to all the other issues that have been posted before?

It seems so, I copied this from https://github.com/fabiangreffrath/woof/commit/6fda34f497da756cad2f50dbefecc7e719fef03a

> Hm, could you maybe integrate this into R_FakeFlat()?

Not sure how to integrate it into `R_FakeFlat()`, but we can simplify a bit:
```diff
diff --git a/src/r_bsp.c b/src/r_bsp.c
index 99c4ecf..f9fe955 100644
--- a/src/r_bsp.c
+++ b/src/r_bsp.c
@@ -189,13 +189,9 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
       // Replace floor and ceiling height with other sector's heights.
       tempsec->floorheight   = s->floorheight;
       tempsec->ceilingheight = s->ceilingheight;
-      tempsec->interpfloorheight   = s->interpfloorheight;
-      tempsec->interpceilingheight = s->interpceilingheight;
 
       // killough 11/98: prevent sudden light changes from non-water sectors:
       if (underwater && (tempsec->  floorheight = sec->floorheight,
-			 tempsec->interpfloorheight = sec->interpfloorheight,
-			 tempsec->interpceilingheight = s->interpfloorheight-1,
 			 tempsec->ceilingheight = s->floorheight-1, !back))
         {                   // head-below-floor hack
           tempsec->floorpic    = s->floorpic;
@@ -207,7 +203,6 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
             if (s->ceilingpic == skyflatnum)
               {
                 tempsec->floorheight   = tempsec->ceilingheight+1;
-                tempsec->interpfloorheight = tempsec->interpceilingheight+1;
                 tempsec->ceilingpic    = tempsec->floorpic;
                 tempsec->ceiling_xoffs = tempsec->floor_xoffs;
                 tempsec->ceiling_yoffs = tempsec->floor_yoffs;
@@ -236,8 +231,6 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
           {   // Above-ceiling hack
             tempsec->ceilingheight = s->ceilingheight;
             tempsec->floorheight   = s->ceilingheight + 1;
-            tempsec->interpceilingheight = s->interpceilingheight;
-            tempsec->interpfloorheight   = s->interpceilingheight + 1;
 
             tempsec->floorpic    = tempsec->ceilingpic    = s->ceilingpic;
             tempsec->floor_xoffs = tempsec->ceiling_xoffs = s->ceiling_xoffs;
@@ -246,7 +239,6 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
             if (s->floorpic != skyflatnum)
               {
                 tempsec->ceilingheight = sec->ceilingheight;
-                tempsec->interpceilingheight = sec->interpceilingheight;
                 tempsec->floorpic      = s->floorpic;
                 tempsec->floor_xoffs   = s->floor_xoffs;
                 tempsec->floor_yoffs   = s->floor_yoffs;
@@ -262,6 +254,8 @@ sector_t *R_FakeFlat(sector_t *sec, sector_t *tempsec,
               *ceilinglightlevel = s->ceilinglightsec == -1 ? s->lightlevel :
               sectors[s->ceilinglightsec].lightlevel; // killough 4/11/98
           }
+      tempsec->interpfloorheight   = s->interpfloorheight;
+      tempsec->interpceilingheight = s->interpceilingheight;
       sec = tempsec;               // Use other sector
     }
   return sec;
```

Comment (fabiangreffrath) [reply src/r_bsp.c:386]: Okay, then it's good. 😁

**Pull request merged**